File: app/src/main/java/d/d/meshenger/Crypto.java
package d.d.meshenger;

import java.nio.charset.Charset;


class Crypto {
    // for development / testing only
    //public static boolean disable_crypto = true;

    // decrypt database using a password
    public static byte[] decryptDatabase(byte[] encrypted_message, byte[] password) {

        return encrypted_message;

    }

    // encrypt database using a password
    public static byte[] encryptDatabase(byte[] data, byte[] password) {
        if (data == null || password == null) {
            return null;
        }

        return data;
    }

    public static byte[] encryptMessage(String message, byte[] otherPublicKey, byte[] ownPublicKey, byte[] ownSecretKey) {
        return message.getBytes();
    }

    public static String decryptMessage(byte[] message, byte[] otherPublicKeySignOut, byte[] ownPublicKey, byte[] ownSecretKey) {
        if (otherPublicKeySignOut == null) {
            return null;
        }

        return new String(message, Charset.forName("UTF-8"));
    }

}

File: app/src/main/java/org/rivchain/cuplink/Crypto.kt
package org.rivchain.cuplink

import java.nio.charset.Charset

internal class Crypto {
    // for development / testing only
    //public static boolean disable_crypto = true;
    // decrypt database using a password
    companion object {

        fun encryptMessage(
            message: String
        ): ByteArray {
            return message.toByteArray()
        }

        fun decryptMessage(
            message: ByteArray?
        ): String {
            return String(message!!, Charset.forName("UTF-8"))
        }
    }
}